src/net.cpp

```diff
@@ -667,6 +676,19 @@ bool CNode::ReceiveMsgBytes(const char *pch, unsigned int nBytes)
         nBytes -= handled;
 
         if (msg.complete()) {
+
+            //store received bytes per command
+            //to prevent a memory DOS, only allow valid commands
+            std::string command(msg.hdr.pchCommand);
+            for (unsigned int i = 0; i < sizeof(logAllowIncommingCmds)/sizeof(logAllowIncommingCmds[0]); i++)
```

Comment:
Mostly I was thinking less code/easier to read (though it would be faster too, since we have to do a find() anyway).